src/scripts.rs

```diff
@@ -274,6 +279,9 @@ macro_rules! deposit_desc_checks {
         if $stakeholders.len() < 2 {
             return Err(ScriptCreationError::BadParameters);
         }
+        if $stakeholders.len() > MAX_STAKEHOLDERS {
+            return Err(ScriptCreationError::PolicyCompilation(CompilerError::LimitsExceeded));
```

Comment:
We shouldn't return rust-miniscript's error here. It's ugly and it's not the right meaning: "limits exceeded" is for Bitcoin's resource limits.
https://github.com/rust-bitcoin/rust-miniscript/blob/6cd1fb66241b08439d8ad2d1335828007519c204/src/policy/compiler.rs#L57-L61

Here it's our own bounds so `ScriptCreationError::BadParameters` would be more appropriate.